analytics.js

```diff
@@ -647,7 +647,7 @@ class Analytics {
     if (userId && this.userId && userId !== this.userId) {
       this.reset();
     }
-    this.userId = userId;
+    this.userId = typeof userId === 'string' ? userId : userId.toString();
```

Comment:
The simple `toString` may not always work. Use the stringificant-parse logic used in cookies here as well.

Reply:
Agreed that toString might not be reliable. Changed it to JSON.stringify(). But parsing is not required.